zypp/KeyRing.h

```diff
@@ -96,6 +98,23 @@ namespace zypp
      */
     virtual bool askUserToAcceptVerificationFailed( const std::string &file, const PublicKey &key, const KeyContext &keycontext = KeyContext() );
 
+    /**
+     * Ask user to trust and/or import the package key to trusted keyring, using ReportBase::report
+     *
+     * The UserData object will have the following fields:
+     * "PublicKey"		The PublicKey to be accepted
+     * "KeyContext"		The KeyContext
+     *
+     * Userdata accepted:
+     * "TrustKey"			bool user can either trust or not trust the key
+     *
+     * \see KeyTrust
+     * \sa ReportBase::report
+     * \note this is a non virtual function and will use ReportBase::report to send the report.
+     *
+     */
+    bool askUserToAcceptPackageKey( const PublicKey &key, const KeyContext &keycontext = KeyContext() );
```

Comment:
`KeyRing.h/.cc` separate commit introducing the callback

Reply:
Done